symtabAPI/src/emitElfStatic-riscv64.C

```diff
@@ -0,0 +1,213 @@
+ * the function interface are explicitly left out.
+ */
+
+bool emitElfStatic::archSpecificRelocation(Symtab *, Symtab *, char *, relocationEntry &,
+        Offset, Offset, Offset, LinkMap &, string &) {
+    assert(!EMIT_STATIC_ASSERT);
+    return false;
+}
+
+bool emitElfStatic::checkSpecialCaseSymbols(Symtab *, Symbol *) {
+    return true;
+}
+
+Offset emitElfStatic::layoutTLSImage(Offset, Region *, Region *, LinkMap &) {
+    assert(!EMIT_STATIC_ASSERT);
+    return 0;
+}
+
+Offset emitElfStatic::adjustTLSOffset(Offset, Offset) {
+    assert(!EMIT_STATIC_ASSERT);
+    return 0;
+}
+
+char emitElfStatic::getPaddingValue(Region::RegionType) {
+    assert(!EMIT_STATIC_ASSERT);
+    return 0;
+}
+
+void emitElfStatic::cleanupTLSRegionOffsets(map<Region *, LinkMap::AllocPair> &,
+        Region *, Region *)
+{
+    assert(!EMIT_STATIC_ASSERT);
+}
+
+bool emitElfStatic::isGOTRelocation(unsigned long) {
+    assert(!EMIT_STATIC_ASSERT);
+    return false;
+}
+
+#ifdef OLD_VERSION
+Offset emitElfStatic::getGOTSize(LinkMap &) {
+    assert(!EMIT_STATIC_ASSERT);
+    return 0;
+}
+#endif
+
+Offset emitElfStatic::getGOTSize(Symtab *, LinkMap &, Offset &) {
+    assert(!EMIT_STATIC_ASSERT);
+    return 0;
+}
+
+Offset emitElfStatic::getGOTAlign(LinkMap &) {
+    assert(!EMIT_STATIC_ASSERT);
+    return 0;
+}
+
+#ifdef OLD_VERSION
+void emitElfStatic::buildGOT(LinkMap &) {
+    assert(!EMIT_STATIC_ASSERT);
+}
+#endif
+
+void emitElfStatic::buildGOT(Symtab *, LinkMap &) {
+    assert(!EMIT_STATIC_ASSERT);
+}
+
+bool emitElfStatic::isConstructorRegion(Region *) {
+    assert(!EMIT_STATIC_ASSERT);
+    return false;
+}
+
+Offset emitElfStatic::layoutNewCtorRegion(LinkMap &) {
+    assert(!EMIT_STATIC_ASSERT);
+    return 0;
+}
+
+bool emitElfStatic::createNewCtorRegion(LinkMap &) {
+    assert(!EMIT_STATIC_ASSERT);
+    return false;
+}
+
+bool emitElfStatic::isDestructorRegion(Region *) {
+    assert(!EMIT_STATIC_ASSERT);
+    return false;
+}
+
+bool emitElfStatic::isGOTRegion(Region *) {
+    assert(!EMIT_STATIC_ASSERT);
+    return false;
+}
+
+Offset emitElfStatic::layoutNewDtorRegion(LinkMap &) {
+    assert(!EMIT_STATIC_ASSERT);
+    return 0;
+}
+
+bool emitElfStatic::createNewDtorRegion(LinkMap &) {
+    assert(!EMIT_STATIC_ASSERT);
+    return false;
+}
+
+void emitElfStatic::getExcludedSymbolNames(set<string> &) {
+    assert(!EMIT_STATIC_ASSERT);
+}
+
+Offset emitElfStatic::allocStubRegions(LinkMap &, Offset){
+    assert(!EMIT_STATIC_ASSERT);
+    return 0;
+}
+
+bool emitElfStatic::updateTOC(Symtab *, LinkMap &, Offset){
+    assert(!EMIT_STATIC_ASSERT);
+    return false;
+}
+
+inline
+static bool adjustValInRegion(Region *reg, Offset offInReg, Offset addressWidth, int adjust) {
+    Offset newValue;
+    unsigned char *oldValues;
+
+    oldValues = reinterpret_cast<unsigned char *>(reg->getPtrToRawData());
+    memcpy(&newValue, &oldValues[offInReg], addressWidth);
```

Comment:
This assumes an `Offset` can hold `addressWidth` bytes. That seems like a weird constraint, but is probably one of those implicit, unsubstantiated assumptions Dyninst uses all the time.

Reply:
I believe so